src/wasm/request.rs

```diff
@@ -3,6 +3,7 @@ use std::fmt;
 
 use http::Method;
 use url::Url;
+pub use web_sys::{RequestMode, RequestCache};
```

Comment:
Here `web_sys::RequestMode` and `web_sys::RequestCache` are re-exported. I'm not confident to expose `web_sys` things as part of APIs provided by this crate. If we should make our own enum for these modes like below, please let me know. I'll update this patch.

```rust
#[derive(Debug)]
pub enum RequestFetchMode {
    Cors,
    NoCors,
    SameOrigin,
    Navigate,
}
```

Reply:
Ya, it's probably best to make the API make sense with and without WASM. Like if reqwest were to eventually add cache support.

Is there a specific mode you need right now, or just wanting to add general support?

Reply:
Alternatively to exposing the actual enums, you could provide specific functions on the builder that use these enums internally (names subject to bike shedding):

```rust
fn fetch_mode_cors(mut self) -> RequestBuilder;
fn fetch_mode_no_cors(mut self) -> RequestBuilder;
fn fetch_mode_same_origin(mut self) -> RequestBuilder;
...
```

This is more verbose but avoids leaking any underlying types (for now).

Reply:
> Is there a specific mode you need right now, or just wanting to add general support?

Currently I need `no-cors` mode. Cache mode is not immediate for me (I just added the API in addition to request fetch mode for general support).

AFAIK CORS is a mechanism only implemented in browser. In terms of aligning APIs with and without Wasm, how about defining general 'request mode' API and defining Wasm-specific variants in it? (it's just an idea, tho)

```rust
#[cfg(not(target_arch = "wasm32"))]
pub enum RequestMode {
    Default,
    // Currently empty, but when some fetch behaviors are implemented as mode,
    // they will go here
}

#[cfg(target_arch = "wasm32")]
pub enum RequestMode {
    Default,
    Cors,
    NoCors,
    SameOrigin,
}
```

Reply:
How about let's start conservative and add `fetch_mode_no_cors()`, which in not-WASM just does nothing.

Reply:
It also looks fine for me. I'll update this branch.